msgpack/ext.py

```diff
@@ -157,12 +157,14 @@ def to_datetime(self):
         :rtype: `datetime.datetime`
         """
         utc = datetime.timezone.utc
-        return datetime.datetime.fromtimestamp(0, utc) + datetime.timedelta(seconds=self.to_unix())
+        return datetime.datetime.fromtimestamp(0, utc) + datetime.timedelta(
+            seconds=self.seconds, microseconds=round(self.nanoseconds / 1e3)
+        )
 
     @staticmethod
     def from_datetime(dt):
         """Create a Timestamp from datetime with tzinfo.
 
         :rtype: Timestamp
         """
-        return Timestamp.from_unix(dt.timestamp())
+        return Timestamp(seconds=int(dt.timestamp() // 1), nanoseconds=dt.microsecond * 10**3)
```

Comment:
> That x // 1 feels strange.
> 
> Considering you use int() after that anyway, why do you need that?

Yes, i overlooked that part completely, we dont need the floor division at all. I'll remove it.

> Also: just use 1000 instead of 10**3?

we can of course use 1000 instead, its just that in the code that i checked, i either saw "10**x" or "1ex" and wanted follow that convention.

Reply:
> That x // 1 feels strange.
> 
> Considering you use int() after that anyway, why do you need that?

Yes, i overlooked that part completely, we dont need the floor division at all. I'll remove it.

> Also: just use 1000 instead of 10**3?

we can of course use 1000 instead, its just that in the code that i checked, i either saw "10**x" or "1ex" and wanted follow that convention.